wpimath/src/main/java/edu/wpi/first/math/kinematics/SwerveDriveOdometry.java

```diff
@@ -92,17 +92,34 @@ public Pose2d getPoseMeters() {
    */
   public Pose2d updateWithTime(
       double currentTimeSeconds, Rotation2d gyroAngle, SwerveModuleState... moduleStates) {
+    return updateWithTime(
+        currentTimeSeconds, gyroAngle, m_kinematics.toChassisSpeeds(moduleStates));
+  }
+
+  /**
+   * Updates the robot's position on the field using forward kinematics and integration of the pose
+   * over time. This method takes in the current time as a parameter to calculate period (difference
+   * between two timestamps). The period is used to calculate the change in distance from a
+   * velocity. This also takes in an angle parameter which is used instead of the angular rate that
+   * is calculated from forward kinematics.
+   *
+   * @param currentTimeSeconds The current time in seconds.
+   * @param gyroAngle The angle reported by the gyroscope.
+   * @param chassisSpeeds The state of the swerve modules after forward kinematics.
+   * @return The new pose of the robot.
+   */
+  public Pose2d updateWithTime(
```

Comment:
There's nothing that is swerve-specific in this method anymore. Maybe we should create a generic `Odometry` class and add this method there? That way, people with unconventional drive systems (e.g. H-Drive) can also use the `update` method that takes in `ChassisSpeeds`?

Maybe @Oblarg or @calcmogul have more thoughts on this.

Reply:
Makes sense to me. This whole class could be renamed to Odometry, then classes like SwerveDriveOdometry would inherit from it and add a kinematics class member variable and their custom update() and updateWithTime() functions.